lib/base/dictionary.cpp

```diff
@@ -263,6 +267,7 @@ String Dictionary::ToString() const
 void Dictionary::Freeze()
 {
 	ObjectLock olock(this);
+	std::unique_lock<std::shared_timed_mutex> lock (m_DataMutex);
 	m_Frozen = true;
 }
```

Comment:
Why is this needed here?

Reply:
Wasn't it your idea to double-lock all modificators?

Reply:
Everything that modifies `m_Data`.

In `Namespace`, the idea of `m_DataMutex` is to only protect only the integrity of the `m_Data` data structure itself (hence the name) and indeed, `Namespace::Freeze()` doesn't acquire `m_DataMutex`.

Here, `m_Frozen` is still fully synchronized using the `ObjectLock` (only write operations read it, these also obtain the `ObjectLock`). In `Namespace`, additionally read operations check `m_Frozen` to disable locking, that's why it was made a `std::atomic<bool>` there.